src/main/java/com/aws/greengrass/util/platforms/unix/LinuxPlatform.java

```diff
@@ -0,0 +1,143 @@
+    private static final String CGROUP_MEMORY_LIMITS = "memory.limit_in_bytes";
+    private static final String CPU_KEY = "CPU";
+    private static final String UNICODE_SPACE = "\\040";
+    private static final List<LinuxCgroup> ENABLED_CGROUPS = Arrays.asList(LinuxCgroup.Memory, LinuxCgroup.CPU);
+
+    public void createResourceManager(String componentName) {
+        try {
+            Set<String> mounts = getMountedPaths();
+            if (!mounts.contains(LinuxCgroup.getRootPath())) {
+                runCmdWithRoot(LinuxCgroup.rootMountCmd());
+                for (LinuxCgroup cg : ENABLED_CGROUPS) {
+                    Files.createDirectory(cg.getSubsystemRootPath());
+                }
+            }
+
+            for (LinuxCgroup cg : ENABLED_CGROUPS) {
+                if (!mounts.contains(cg.getSubsystemRootPath())) {
+                    runCmdWithRoot(cg.subsystemMountCmd());
+                }
+                if (!Files.exists(cg.getSubsystemGGPath())) {
+                    Files.createDirectory(cg.getSubsystemGGPath());
+                }
+                if (!Files.exists(cg.getSubsystemComponentPath(componentName))) {
+                    Files.createDirectory(cg.getSubsystemComponentPath(componentName));
+                }
+            }
+        } catch (IOException e) {
+            logger.atError().setCause(e).kv("componentName", componentName)
+                    .log("Failed to create system resource manager");
+        } catch (InterruptedException e) {
+            logger.atWarn("Got interrupted");
+        }
+    }
+
+    public void applyResourceLimits(String componentName, Map<String, Object> resourceLimit) {
+        try {
+            if (resourceLimit.containsKey(MEMORY_KEY)) {
+                long memoryLimitInKB = (long) resourceLimit.get(MEMORY_KEY);
+
+                // TODO: add input validation
+                String memoryLimit = Long.toString(memoryLimitInKB * 1024);
+                Files.write(LinuxCgroup.Memory.getSubsystemComponentPath(componentName).resolve(CGROUP_MEMORY_LIMITS),
+                        memoryLimit.getBytes());
+            }
+
+            if (resourceLimit.containsKey(CPU_KEY)) {
+                double cpu = (double) resourceLimit.get(CPU_KEY);
+
+                byte[] content = Files.readAllBytes(
+                        LinuxCgroup.CPU.getSubsystemComponentPath(componentName).resolve("cpu.cfs_period_us"));
+                int cpuPeriodUs = Integer.parseInt(new String(content));
+
+                int cpuQuotaUs = (int) (cpuPeriodUs * cpu);
+                String cpuQuotaUsStr = Integer.toString(cpuQuotaUs);
+
+                Files.write(LinuxCgroup.CPU.getSubsystemComponentPath(componentName).resolve("cpu.cfs_quota_us"),
+                        cpuQuotaUsStr.getBytes());
+            }
+
+        } catch (IOException e) {
+            logger.atError().setCause(e).kv("componentName", componentName)
+                    .log("Failed to apply resource limits");
```

Comment:
Same here and line 100, only logging may not be sufficient for failure handling?